### PR TITLE
upgpatch: linux 6.0.2.arch1-1

### PR DESCRIPTION
**Pull request by Xeonacid**
1. Backport [RISC-V: Make mmap() with PROT_WRITE imply PROT_READ](https://git.kernel.org/pub/scm/linux/kernel/git/riscv/linux.git/commit/?h=for-next&id=8aeb7b17f04ef40f620c763502e2b644c5c73efd).
2. Force disable sv57, since many packages ]like go and openjdk cannot handle it.
3. Disable VMAP_STACK, or the kernel will crash on Unmatched with UEFI. [Ubuntu bug report](https://bugs.launchpad.net/ubuntu/+source/linux-riscv/+bug/1967130)

Note for reviewers: [Arch's `prepare()`](https://github.com/archlinux/svntogit-packages/blob/packages/linux/repos/core-x86_64/PKGBUILD#L43-L50) will apply all patches with 	`.patch` extension, so we don't need to patch manually.

### CODE REVIEW

File: linux/riscv64.config-patch [hunk 1]
```diff
@@ -54,15 +54,15 @@
  # end of RISC-V CPU Idle Drivers
  # end of CPU Idle
  # end of CPU Power Management
-@@ -1737,6 +1744,7 @@
- CONFIG_PCI_STUB=y
- CONFIG_PCI_PF_STUB=m
- CONFIG_PCI_ATS=y
-+CONFIG_PCI_ECAM=y
```
Comment (XieJiSS) [inline linux/riscv64.config-patch:61]: Is the removal of `+CONFIG_PCI_ECAM=y` intended?

File: linux/riscv64.config-patch [hunk 2]
```diff
@@ -206,17 +206,7 @@
  CONFIG_DMADEVICES=y
  # CONFIG_DMADEVICES_DEBUG is not set
  
-@@ -7822,7 +7846,8 @@
- #
- # VME Device Drivers
- #
--# CONFIG_GOLDFISH is not set
```
Comment (XieJiSS) [inline linux/riscv64.config-patch:213]: ditto

File: linux/riscv64.config-patch [hunk 3]
```diff
@@ -206,17 +206,7 @@
...
-+CONFIG_GOLDFISH=y
-+# CONFIG_GOLDFISH_PIPE is not set
```
Comment (XieJiSS) [inline linux/riscv64.config-patch:215]: ditto

File: linux/riscv64.config-patch [hunk 4]
```diff
@@ -206,17 +206,7 @@
  CONFIG_DMADEVICES=y
  # CONFIG_DMADEVICES_DEBUG is not set
  
-@@ -7822,7 +7846,8 @@
- #
- # VME Device Drivers
- #
--# CONFIG_GOLDFISH is not set
-+CONFIG_GOLDFISH=y
```
Comment (XieJiSS) [inline linux/riscv64.config-patch:214]: ditto

File: linux/riscv64.config-patch (hunk 1 above)
Comment (Xeonacid) [reply linux/riscv64.config-patch:61]: This is modified by `make olddefconfig` and seems almost unrelated to riscv.

**Review by XieJiSS — APPROVED** [review]
LGTM

**Pull request merged**